server/routes/auth.js

```diff
@@ -4,17 +4,23 @@ import AuthController from '../controllers/AuthController';
 import middlewares from '../middlewares';
 
 const { verifyToken, UserValidator } = middlewares;
-
 const auth = express.Router();
 const BASE_URL = '/auth';
-const { createUser, login, verifyUser } = AuthController;
-const { signUpValidation, loginValidation } = UserValidator;
-
+const {
+  createUser, login, verifyUser, resetpasswordEmail, resetPassword
+} = AuthController;
+const {
+  signUpValidation, loginValidation, PasswordValidation, EmailValidation
+} = UserValidator;
 // Sign up route
 auth.post(`${BASE_URL}/signup`, signUpValidation(), createUser);
-// Login route
+/* Login route */
 auth.post(`${BASE_URL}/login`, loginValidation(), login);
-// Verify User route
+/* Verify User route */
 auth.get(`${BASE_URL}/verify`, verifyToken, verifyUser);
+/* route responsible for sending password reset link to owners mail */ 
+auth.post(`${BASE_URL}/passwordreset`, EmailValidation(), resetpasswordEmail);
+/* route responsible for password reset when mail has been sent */
+auth.patch(`${BASE_URL}/passwordresetVerify`, verifyToken, PasswordValidation(), resetPassword);
```

Comment:
Great job. So I tried to test it on m local before it. Here is what I did
* I signed up with my email
* I visited `POST api/v1/auth/passwordreset` with my mail
* I changed the the heroku url on the email sent to local host so I can test on my local
* an email was sent to me
* on clicking the link, I go a 404 not found error
* Tried to find out what the problem was, and I discovered this is a patch request. 
So my question
 * How is the user going to reset their password through the link if it is sending a patch request? keeping in mind that the link from their email will make a get request when they click it.